### DISCUSSION TITLE
Wield, sometimes my hero have no face, sometimes have, how can i make sure of it ?

### DISCUSSION THREAD

**Discussion by ndkk** [flareteam/flare-game#944] [Q&A] [answered]
Sometimes my hero have no face, sometimes have.
I don't know why.  

as below:
[https://github.com/ndkk/uploadfile/blob/main/noface.png](https://github.com/ndkk/uploadfile/blob/main/noface.png)

Maybe this is a bug?

My tile is home.tmx  : 
[https://github.com/ndkk/hero/blob/main/tiled/geos/home.tmx](https://github.com/ndkk/hero/blob/main/tiled/geos/home.tmx)

Not only this scene, but also many scenes.

**Comment by CaptainFluffy1903** [comment]
Does this only happen when character face to one direction?

**Comment by CaptainFluffy1903** [comment]
<img width="482" alt="Ekran görüntüsü 2023-06-16 133628" src="https://github.com/flareteam/flare-game/assets/122923399/c8fc0c09-332a-46e7-b0a6-7d2f3fad5f99">
I downloaded ur mod and there wasnt any problem.

**Comment by CaptainFluffy1903** [comment]
Take a look to character animations and then size and layer of ur tiles.This issue doesnt happen when ı tried it.

**Comment by CaptainFluffy1903** [comment]
Its weird, I'm still playing ur mod but doesnt see any issue like that.

**Reply by ndkk** [reply]
but i encountered this issue all the time. :disappointed:

**Comment by ndkk** [comment]
not sure, there are two directions:
[https://github.com/ndkk/uploadfile/blob/main/nobody.png](https://github.com/ndkk/uploadfile/blob/main/nobody.png)
[https://github.com/ndkk/uploadfile/blob/main/nobody2.png](https://github.com/ndkk/uploadfile/blob/main/nobody2.png)

**Comment by ndkk** [comment]
my environment:

uname -a
Linux ubuntu 5.15.0-73-generic #80~20.04.1-Ubuntu SMP Wed May 17 14:58:14 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

**Comment by dorkster** [comment, marked as answer]
This is a draw order issue similar to the issue you had with tall tiles. Entities like the player are rendered in the same pass as the tiles on the layer named "object". This happens top-to-bottom, left-to-right. So in your screenshot, the rug tile is to the right of the player, and gets drawn on top as a result.

The solution is to create another layer between the "background" and "object" layers. I'll usually call this the "fringe" layer. Put the rug tiles on the fringe layer, and they'll then always render behind entities.

**Reply by ndkk** [reply]
Thanks !  I get it ! 
Actually the difference between hide and not hide is difficult!
